src/coreclr/gc/gc.cpp

```diff
@@ -19469,6 +19472,30 @@ bool gc_heap::init_table_for_region (int gen_number, heap_segment* region)
 }
 #endif //USE_REGIONS
 
+BOOL gc_heap::consumed_enough_gen2_budget_p (dynamic_data* dd_max, uint32_t memory_load)
+{
+    // we computed gen2 budget under some memory load - check if the memory load is significantly higher now
+    uint32_t prev_remaining_memory = 100 - dd_entry_memory_load (dd_max);
+    uint32_t curr_remaining_memory = 100 - memory_load;
+
+    // if the previous memory load was already at the limit, the situation cannot have gotten worse
+    if (prev_remaining_memory <= 0)
```

Comment:
>   if (prev_remaining_memory <= 0) [](http://example.com/codeflow?start=1&length=34)

how can this ever be < 0?

Reply:
Abundance of caution - some operating system _might_ return a value > 100 for memory load - we want to treat this as 100...

Reply:
okie but this is a uint32_t, not an int32_t :)